privacyidea.py

```diff
@@ -119,47 +150,47 @@ def authenticate(self, configurationAttributes, requestParameters, step):
             fm.add(FacesMessage.SEVERITY_ERROR, "Failed to communicate to privacyIDEA. Possible misconfiguration. Please have the administrator check the log files.")
             return False
 
-        authenticationService = CdiUtil.bean(AuthenticationService)
         identity = CdiUtil.bean(Identity)
 
         if step == 1:
             credentials = identity.getCredentials()
-            user_name = credentials.getUsername()
-            user_password = credentials.getPassword()
-
-            if user_name:
-                self.addToSession("currentUser", user_name)
-
-                response = self.pi.validateCheck(user_name, user_password)
-                if response:
-                    # First check if what was entered is sufficient to log in
-                    if response.getValue():
-                        logged_in = authenticationService.authenticate(user_name)
-                        if logged_in:
-                            self.addToSession("auth_success", True)
-                        return logged_in
-
-                    # If not, check if transaction was triggered
+            username = credentials.getUsername()
+            password = credentials.getPassword()
+
+            if username:
+                if self.sendPassword:
+                    response = self.pi.validateCheck(username, password)
```

Comment:
I thought we still have to ``addToSession(currentUser...``, don't we?

Reply:
see the comment below